src/main/rp/ReplayRegistry.ts

```diff
@@ -0,0 +1,126 @@
+import { AuthorizationRequest } from '../authorization-request';
+import { AuthorizationResponse } from '../authorization-response';
+import { RequestObjectPayload } from '../types';
+import { AuthorizationEvents } from '../types/Events';
+import {
+  AuthorizationRequestState,
+  AuthorizationRequestStateStatus,
+  AuthorizationResponseState,
+  AuthorizationResponseStateStatus,
+} from '../types/ReplayRegistry';
+
+import { rpEventEmitter } from './RP';
+
+export class ReplayRegistry {
+  private authorizationRequests: Map<string, AuthorizationRequestState> = new Map<string, AuthorizationRequestState>();
+  private authorizationResponses: Map<string, AuthorizationResponseState> = new Map<string, AuthorizationResponseState>();
+
+  public constructor() {
```

Comment:
if you have the emitter as constructor argument, you can have multiple replay registries (one per RP)

Reply:
made some changes